### PR TITLE
Feature/javadoc

### PR DESCRIPTION
**Pull request by Di3GO95**
Javadoc and bean validation

### CODE REVIEW

File: src/main/java/org/diego/tutorial/car/resources/CarResource.java [hunk 1]
```diff
@@ -48,8 +60,22 @@ public Response getCars(@QueryParam("country") String country){
...
 	public Response addCar(Car car) {
+		List<String> validationErrors = CarValidator.validateAddAndUpdate(car);
+		if (!validationErrors.isEmpty()) {
+			String message = "Request to add a car with non valid fields: ";
+			for (String validationError : validationErrors) {
+				message += validationError + " - ";
+			}
+			throw new BadRequestException(message);
+		}
+
```
Comment (javieraviles) [inline src/main/java/org/diego/tutorial/car/resources/CarResource.java]: this should be encapsulated into some method somewhere else

File: src/main/java/org/diego/tutorial/car/resources/CarResource.java [hunk 2]
```diff
@@ -74,11 +109,27 @@ public Response getCar(@PathParam("id") long id) {
...
+		
+		List<String> validationErrors = CarValidator.validateAddAndUpdate(car);
+		if (!validationErrors.isEmpty()) {
+			String message = "Request to update car with non valid fields: <p>";
+			for (String validationError : validationErrors) {
+				message += validationError + "<p>";
+			}
+			throw new BadRequestException(message);
+		}
+
```
Comment (javieraviles) [inline src/main/java/org/diego/tutorial/car/resources/CarResource.java]: this should be encapsulated into some method somewhere else

File: src/main/java/org/diego/tutorial/car/resources/CarResource.java (hunk 1 above)
Comment (Di3GO95) [reply src/main/java/org/diego/tutorial/car/resources/CarResource.java]: Updated with encapsulation methods.

File: src/main/java/org/diego/tutorial/car/resources/CarResource.java (hunk 2 above)
Comment (Di3GO95) [reply src/main/java/org/diego/tutorial/car/resources/CarResource.java]: Updated with encapsulation methods.

**Pull request merged**